File: 3952-trionic-array-i/trionic-array-i.java
class Solution {
    public boolean isTrionic(int[] nums) {
        int n = nums.length;

        // First increasing sequence must exist
        if (nums[1] <= nums[0]) return false;

        int i = 2;
        
        // First strictly increasing phase
        while (i < n && nums[i] > nums[i - 1]) i++;
        if (i == n) return false;

        // Strictly decreasing phase
        while (i < n && nums[i] < nums[i - 1]) i++;
        if (i == n) return false;

        // Final strictly increasing phase
        for (int k = i; k < n; k++) {
            if (nums[k] <= nums[k - 1]) return false;
        }

        return true;
    }
}

File: 3952-trionic-array-i/trionic-array-i.py
class Solution:
    def isTrionic(self, nums: List[int], k: int = 3) -> bool:
        # At beginning nums should be increasing
        if nums[0] >= nums[1]:
            return False

        n = len(nums)

        sign, switches = 1, 0
        for i in range(1, n):
            delta = nums[i] - nums[i - 1]
            if delta == 0:  # Delta should be always decreasing or increasing
                return False
            elif delta * sign < 0:  # Switch point detected
                sign *= -1
                switches += 1

        return switches == k - 1